iceoryx_binding_c/include/iceoryx_binding_c/listener.h

```diff
@@ -46,6 +46,21 @@ ENUM iox_ListenerResult iox_listener_attach_subscriber_event(iox_listener_t cons
                                                              const ENUM iox_SubscriberEvent subscriberEvent,
                                                              void (*callback)(iox_sub_t));
 
+/// @brief Attaches a subscriber event to the listener. The callback has an additional contextData argument to provide
+/// access to user defined data.
+/// @param[in] self listener to which the event should be attached to
+/// @param[in] subscriber subscriber which emits the event
+/// @param[in] subscriberEvent the event which should trigger the listener
+/// @param[in] callback the callback which is called when an event triggers the listener
+/// @param[in] contextData a void pointer which is provided as second argument to the callback
```

Comment:
Not sure whether contextData is the best name. Is it really a "context"? As generic as arg is, it is the argument of the callback right? What about callbackArg? Does not matter much though, it is understandable for those familiar with generic callbacks.

Reply:
The previous name was `userdata` or `uservalue` but then the suggestion was to call it `contextData`. 
 I am also not perfectly happy with that name but I have also no better suggestion.